src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Trait that allows a generic method to find and sort service by priority option in the tag.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+trait PriorityTaggedServiceTrait
+{
+    /**
+     * Finds all services with the given tag name and order them by their priority.
+     *
+     * @param string           $tagName
+     * @param ContainerBuilder $container
+     *
+     * @return Reference[]
+     */
+    private function findAndSortTaggedServices($tagName, ContainerBuilder $container)
+    {
+        $services = $container->findTaggedServiceIds($tagName);
+
+        $sortedServices = array();
+        foreach ($services as $serviceId => $tags) {
```

Comment:
I have to admit that I've not used it before, but it looks interesting.
I'll check it out next week, thanks!

On Fri, 8 Apr 2016 16:32 Javier Eguiluz, <notifications@github.com> wrote:

> In
> src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php
> <https://github.com/symfony/symfony/pull/18482#discussion_r59033296>:
>
> > +trait PriorityTaggedServiceTrait
> > +{
> > +    /**
> > +     * Finds all services with the given tag name and order them by their priority.
> > +     *
> > +     * @param string           $tagName
> > +     * @param ContainerBuilder $container
> > +     *
> > +     * @return Reference[]
> > +     */
> > +    private function findAndSortTaggedServices($tagName, ContainerBuilder $container)
> > +    {
> > +        $services = $container->findTaggedServiceIds($tagName);
> > +
> > +        $sortedServices = array();
> > +        foreach ($services as $serviceId => $tags) {
>
> Could we use SplPriorityQueue
> <http://php.net/manual/en/class.splpriorityqueue.php>?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/symfony/symfony/pull/18482/files/3eb300496ea0c239dae35520d174830cef878e71#r59033296>
>